lib/build.js

```diff
@@ -152,24 +154,35 @@ class BuildModel extends BaseModel {
     start() {
         // Make sure that a pipeline and job is associated with the build
         return this.job
-            .then(job =>
-                job.pipeline.then(pipeline =>
+            .then((job) => {
+                let buildTimeout = DEFAULT_BUILD_TIMEOUT;
+
+                if (this[executor].kubernetes && this[executor].kubernetes.buildTimeout) {
+                    buildTimeout = this[executor].kubernetes.buildTimeout * 60;
+                }
```

Comment:
If buildTimeout is set by cluster config, it will be used.
buildTimeout setting is only set in executor-k8s or executor-k8s-vm

Reply:
I put a comment.